### PR TITLE
Integrate learned literal manager into preprocessing pass context

### PR DESCRIPTION
**Pull request by ajreynol**
Towards "learned rewrite" preprocessing pass / improvements to Int <-> BV.

### CODE REVIEW

File: src/preprocessing/preprocessing_pass_context.h [hunk 1]
```diff
@@ -74,6 +77,18 @@ class PreprocessingPassContext
...
   void recordSymbolsInAssertions(const std::vector<Node>& assertions);
 
+  /**
+   * Notify learned literal. This method is called when a literal is
+   * entailed by the current set of assertions.
+   *
+   * It should be rewritten, and such that top level substitutions have
+   * been applied to it.
+   */
+  void notifyLearnedLiteral(Node lit);
```
Comment (nafur) [inline src/preprocessing/preprocessing_pass_context.h:87]: Use `TNode` instead?

File: src/preprocessing/preprocessing_pass_context.h [hunk 2]
```diff
@@ -57,6 +57,9 @@ class PreprocessingPassContext
     return d_circuitPropagator;
   }
 
+  /** Get the learned literal manager */
+  LearnedLiteralManager* getLearnedLiteralManager();
+
```
Comment (nafur) [inline src/preprocessing/preprocessing_pass_context.h:62]: Do we need this? If so: there is no implementation, and why would we return it as a pointer?

**Review by nafur — CHANGES_REQUESTED** [review]

**Review by yoni206 — COMMENTED** [review]
LGTM, but will let @nafur approve as he had some comments.

File: src/preprocessing/preprocessing_pass_context.h (hunk 2 above)
Comment (ajreynol) [reply src/preprocessing/preprocessing_pass_context.h:62]: Good catch, this was redundant.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**